### PR TITLE
Restructure project

### PR DESCRIPTION
**Pull request by andymiller813**
Large restructure PR to implement the structure mentioned here: https://projectnewmgroup.slack.com/archives/C02T8NE1M08/p1644682503914539

More work to be done especially in the core-theme module and such.  Eventually we will break out more features into separate modules. This pulls all the example KMM code out of the main app

### CODE REVIEW

**Comment by newmskywalker** [comment]
mmm interesting. I wonder why the google-services file didn't get created

**Comment by andymiller813** [comment]
> mmm interesting. I wonder why the google-services file didn't get created

Just fixed it I think. Changed up the directories a tad

File: androidApps/DemoApp/proguard-rules.pro [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (cristhianescobar) [inline androidApps/DemoApp/proguard-rules.pro]: Don't think we need this file

File: androidCore/core-theme/src/main/res/values-night/themes.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<resources xmlns:tools="http://schemas.android.com/tools">
+    <!-- Base application theme. -->
+    <style name="Theme.Newmmobile" parent="Theme.MaterialComponents.DayNight.DarkActionBar">
```
Comment (cristhianescobar) [inline androidCore/core-theme/src/main/res/values-night/themes.xml]: Maybe not as part of this PR, but we should look into define all theming through compose theme.

File: androidFeatures/feature-example/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+android {
+    compileSdk = Versions.compileSdk
+
+    defaultConfig {
+        minSdk = Versions.minSdk
+        targetSdk = Versions.targetSdk
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles("consumer-rules.pro")
```
Comment (cristhianescobar) [inline androidFeatures/feature-example/build.gradle.kts]: Don't think we need like `consumerProguardFiles("consumer-rules.pro")`

File: androidCore/core-theme/src/test/java/io/projectnewm/core/theme/ExampleUnitTest.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package io.projectnewm.core.theme
+
```
Comment (cristhianescobar) [inline androidCore/core-theme/src/test/java/io/projectnewm/core/theme/ExampleUnitTest.kt]: Can we delete this as well

File: androidApps/ProjectNewm/build.gradle.kts [hunk 5]
```diff
@@ -33,17 +33,15 @@ android {
...
     buildFeatures {
         compose = true
     }
+
     composeOptions {
         kotlinCompilerExtensionVersion = Versions.composeCompiler
     }
 
-    hilt {
-        enableTransformForLocalTests = true
```
Comment (cristhianescobar) [inline androidApps/ProjectNewm/build.gradle.kts]: Did we not need this anymore for the UI test we have in place?

File: androidApps/DemoApp/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        targetSdk = Versions.targetSdk
+        versionCode = 1
+        versionName = "1.0"
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = false
+            proguardFiles(
```
Comment (cristhianescobar) [inline androidApps/DemoApp/build.gradle.kts]: We can remove this as well

**Review by cristhianescobar — APPROVED** [review]
💯  Overall, it looks good to me, another thing I would suggest is that now that we have a demo android app and the projectnewm app we can probably split out the work related to the kmm POC 

`Proguard`
I think we can go ahead and remove all Proguard files and references, there isn't  a need for it because of the open source nature of our project. 

`theming`
With the core theme module in place we should be able to get rid of a lot of resource files we used to have, with compose we can bring in all the resources through the theme.

**Comment by cristhianescobar** [comment]
After pulling down the branch and looking at the file structure on Android Studio, I think I would be in favor of nesting `androidApps` `androidCore` `androidFeatures` under one `android`  directory to avoid having unnecessary directories at the top level. Just like, there is one iosApp directory.
I think since we will be sharing  a lot more KMM libraries, I suggest we make those top level. Example, we can have  `feature-api-modules` modules at the top level since they are relevant to both OSs. Same for Core libraries that shared across KMM libraries. Any sort of business logic modules would be good candidates for top level directories.

**Comment by andymiller813** [comment]
> 💯 Overall, it looks good to me, another thing I would suggest is that now that we have a demo android app and the projectnewm app we can probably split out the work related to the kmm POC
> 
> `Proguard` I think we can go ahead and remove all Proguard files and references, there isn't a need for it because of the open source nature of our project.
> 
> `theming` With the core theme module in place we should be able to get rid of a lot of resource files we used to have, with compose we can bring in all the resources through the theme.

I split out the example KMM POC stuff into feature-example already and it's only pulled in by the demo app.  Or was there something else you were referring to? I still need to break it out of the main kmm module for sure.

We are still going to want proguard for shrinking and optimizations, we can just disable the obfuscation piece.

👍 Agreed on theming, we will continue to do more cleanup in future PRs

File: androidApps/ProjectNewm/build.gradle.kts (hunk 5 above)
Comment (andymiller813) [reply androidApps/ProjectNewm/build.gradle.kts]: I got a build warning telling me to remove it 🤷

**Review by newmskywalker — APPROVED** [review]

**Pull request merged**